### PR TITLE
DR-2222 - allow reserved words for synapse columns

### PR DESCRIPTION
**Pull request by snf2ye**

### CODE REVIEW

File: src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java [hunk 1]
```diff
@@ -96,13 +96,13 @@
...
-          + "<columns:{c|<c.name>}; separator=\",\n       \">"
+          + "<columns:{c|[<c.name>]}; separator=\",\n       \">"
           + "<else>"
           + "newid() as datarepo_row_id,\n       "
           + "<columns:{c|"
           + "<if(c.requiresJSONCast)>"
-          + "cast(JSON_VALUE(doc, '$.<c.name>') as <c.synapseDataType>) <c.name>"
-          + "<else>JSON_VALUE(doc, '$.<c.name>') <c.name>"
+          + "cast(JSON_VALUE(doc, '$.[<c.name>]') as <c.synapseDataType>) [<c.name>]"
+          + "<else>JSON_VALUE(doc, '$.[<c.name>]') [<c.name>]"
```
Comment (nmalfroy) [inline src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java]: huh...we need the brackets when pulling out of JSON?  We should make sure it's actually reading out those values as expected

**Review by nmalfroy — APPROVED** [review]
Looks good.  Can you just confirm that it's actually reading the value out in the JSON using that approach?

File: src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java (hunk 1 above)
Comment (snf2ye) [reply src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java]: Should be fixed now.

**Review by fboulnois — APPROVED** [review]
lgtm 👍

**Review by nmalfroy — APPROVED** [review]
thanks for updating the JSON read!

**Pull request merged**